freud/order.pyx

```diff
@@ -598,41 +602,64 @@ cdef class Steinhardt(_PairCompute):
     @property
     def l(self):  # noqa: E743
         """unsigned int: Spherical harmonic quantum number l."""
-        return self.thisptr.getL()
+        ls = self.thisptr.getL()
+        if ls.size() == 1:
+            return ls[0]
+        return ls
 
     @_Compute._computed_property
     def order(self):
-        """float: The system wide normalization of the :math:`q_l` or
-        :math:`w_l` order parameter."""
-        return self.thisptr.getOrder()
+        """float or list of float: The system wide normalization of the
+        :math:`q_l` or :math:`w_l` order parameter for each ``l`` selected. If
+        only 1 ``l`` was selected returns a list of normalizations."""
+        order = self.thisptr.getOrder()
+        if order.size() == 1:
+            return order[0]
+        return order
 
     @_Compute._computed_property
     def particle_order(self):
-        """:math:`\\left(N_{particles}\\right)` :class:`numpy.ndarray`: Variant
-        of the Steinhardt order parameter for each particle (filled with
-        :code:`nan` for particles with no neighbors)."""
-        return freud.util.make_managed_numpy_array(
-            &self.thisptr.getParticleOrder(),
-            freud.util.arr_type_t.FLOAT)
+        """:math:`\\left(N_{particles}\\right)` :class:`numpy.ndarray` or list
+        of `numpy.ndarray`: Variant of the Steinhardt order parameter for each
+        particle (filled with :code:`nan` for particles with no neighbors).
+        Is a list of NumPy arrays if more than one ``l`` was specified."""
+        order_arrays = self.thisptr.getParticleOrder()
+        if order_arrays.size() == 1:
+            return freud.util.make_managed_numpy_array(
+                &order_arrays[0], freud.util.arr_type_t.FLOAT)
+        return [freud.util.make_managed_numpy_array(&order_arrays[i],
+                                                    freud.util.arr_type_t.FLOAT)
+                for i in range(order_arrays.size())]
```

Comment:
Simplify this using a similar construction to the suggestions above. Also applies below.